packages/application-generic/src/usecases/select-integration/select-integration.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { Test } from '@nestjs/testing';
+import { ChannelTypeEnum, EmailProviderIdEnum } from '@novu/shared';
+import {
+  IntegrationEntity,
+  IntegrationRepository,
+  MessageRepository,
+  OrganizationRepository,
+} from '@novu/dal';
+import * as sinon from 'sinon';
+import { UserSession } from '@novu/testing';
+import { SelectIntegration } from './select-integration.usecase';
+import { SelectIntegrationCommand } from './select-integration.command';
+import { GetNovuIntegration } from '../get-novu-integration';
+import { CalculateLimitNovuIntegration } from '../calculate-limit-novu-integration';
+import { AnalyticsService } from '../../services';
+
+describe('select integration', function () {
+  let useCase: SelectIntegration;
+  let integrationRepository: IntegrationRepository;
+  let session: UserSession;
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  beforeEach(async function () {
+    const moduleRef = await Test.createTestingModule({
+      imports: [],
+      providers: [
+        GetNovuIntegration,
+        CalculateLimitNovuIntegration,
+        IntegrationRepository,
+        OrganizationRepository,
+        AnalyticsService,
+        MessageRepository,
+      ],
+    }).compile();
+
+    session = new UserSession();
+    await session.initialize();
+
+    useCase = moduleRef.get<SelectIntegration>(SelectIntegration);
+    integrationRepository = moduleRef.get<IntegrationRepository>(
+      IntegrationRepository
+    );
+  });
+
+  it('should select the integration', async function () {
+    const testIntegration: IntegrationEntity = {
+      _environmentId: 'env-test-123',
+      _id: 'integration-test-123',
+      _organizationId: 'org-test-123',
+      active: true,
+      channel: ChannelTypeEnum.EMAIL,
+      credentials: {
+        apiKey: '123',
+        user: 'test-user',
+        secretKey: '123',
+        domain: 'domain',
+        password: '123',
+        host: 'host',
+        port: 'port',
+        secure: true,
+        region: 'region',
+        accountSid: 'accountSid',
+        messageProfileId: 'messageProfileId',
+        token: '123',
+        from: 'from',
+        senderName: 'senderName',
+        applicationId: 'applicationId',
+        clientId: 'clientId',
+        projectName: 'projectName',
+      },
+      providerId: 'test-provider-id',
+      deleted: false,
+      identifier: 'test-integration-identifier',
+      name: 'test-integration-name',
+      deletedAt: null,
+      deletedBy: null,
+    };
+
+    sinon.stub(integrationRepository, 'findOne').resolves(testIntegration);
```

Comment:
Is the data stubbed because the problems are you finding running the tests in Application Generic?

Reply:
I've updated the tests, they are passing right now